x-pack/plugins/endpoint/public/applications/endpoint/store/selectors.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { GlobalState } from './reducer';
+import * as alertListSelectors from './alerts/selectors';
+
+export const alertListData = composeSelectors(
```

Comment:
FYI. 
@parkiino will be pushing through a PR soon that takes a different approach for using a selector from a React component. Instead of having to re-define all your "context bound" selectors again here so that they receive the correct namespace from the `GlobalStore`, instead we're proposing having a hook like:

File: `store/hooks.ts`:
```javascript
export function useEndpointListSelector<TSelected>(
  selector: (state: EndpointListState) => TSelected
) {
  return useSelector(function(state: GlobalState) {
    return selector(state.endpointList);
  });
}
```

The idea is that each of the concerns will have an associated hook.  To use this in a component, you would:

```jsx
import { endpointListData } from "../store/endpoint_list/selectors;
import { useEndpointListSelector } from "../store/hooks;

const EndpointsListPage = () => {
    const listData = useEndpointListSelector(endpointListData);

    return (<div>...</div>)
}
```

(Credit to @oatkiller for the idea during a recent discussion 😃  )

Reply:
@paul-tavares If we use this approach, how would a middleware (or saga) use the selector if needed to grab some data from state?

Reply:
@peluja1012  - From a saga or middleware we have access directly to the `store` and can get back `GlobalState` - so you would use the concern selector directly and feed it its namespace (ex. `endpointListData(store.getState().endpointList)`).

When we discussed this, the goal was to avoid the need to define all concerns selectors a second time in order for us to continue to use `react-redux` `useSelector()`.  The hook does that nicely 😃